pkgs/development/haskell-modules/lib.nix

```diff
@@ -414,13 +414,13 @@ rec {
   */
   generateOptparseApplicativeCompletion = exeName: pkg: overrideCabal pkg (drv: {
     postInstall = (drv.postInstall or "") + ''
-      bashCompDir="$out/share/bash-completion/completions"
-      zshCompDir="$out/share/zsh/vendor-completions"
-      fishCompDir="$out/share/fish/vendor_completions.d"
+      bashCompDir="''${!outputBin}/share/bash-completion/completions"
```

Comment:
What does `${!outputBin}` do here?  Is it more appropriate than `${out}`?

Reply:
`$outputBin` [is `"bin"` or `"out"`](https://nixos.org/manual/nixpkgs/stable/#multiple-output-file-type-groups).
`${!x}` is a double lookup of `x`, so `${!outputBin}` evaluates to either `$bin` or `$out`.

Reply:
I am sorry, that I ask stupid questions here, but I am curious about that variable and the manual didn’t enlighten me.
I know what `$out` does, but how would `$bin` look like in which kind of derivations is ${outputBin} = $bin?
For my uninformed eyes the $bin/bin/... in this change looks like a potential bug.

Reply:
To clarify: My naive expectations would be that $bin = $out/bin

Reply:
`$bin` is just the `bin` output store path e.g `/nix/store/...-niv-1.0-bin`, `$bin/bin` is the "FHS" bin directory inside the separate bin output `/nix/store/...-niv-1.0-bin/bin`.

I'll also point out that the `bin` output isn't limited to having only a `bin` directory. It makes sense to have supporting files like this in the same output. A `data` or `share` output is not helpful because it's only used in conjunction with `bin`, negating potential closure size benefits and doing so creates a risk of creating a reference cycle in the output, which is not allowed.